bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/protocol/SoulissBindingUDPDecoder.java

```diff
@@ -0,0 +1,609 @@
+    }
+
+    /**
+     * Decodes lower level MaCaCo packet
+     *
+     * @param lastByteGatewayIP
+     *
+     * @param macacoPck
+     */
+    private void decodeMacaco(byte lastByteGatewayIP, ArrayList<Byte> macacoPck) {
+        int functionalCode = macacoPck.get(0);
+        switch (functionalCode) {
+            case SoulissBindingUDPConstants.SOULISS_UDP_FUNCTION_PING_RESP:
+                logger.debug("Received functional code: 0x{}- Ping answer", Integer.toHexString(functionalCode));
+                decodePing(lastByteGatewayIP, macacoPck);
+                break;
+            case SoulissBindingUDPConstants.SOULISS_UDP_FUNCTION_DISCOVER_GW_NODE_BCAST_RESP:
+                logger.debug("Received functional code: 0x{} - Discover a gateway node answer (broadcast)",
+                        Integer.toHexString(functionalCode));
+                try {
+                    decodePingBroadcast(macacoPck);
+                } catch (UnknownHostException e) {
+                    logger.debug("Error: {}", e.getLocalizedMessage());
+                    logger.error("Error:", e);
+                }
+                break;
+
+            case SoulissBindingUDPConstants.SOULISS_UDP_FUNCTION_SUBSCRIBE_RESP:
+            case SoulissBindingUDPConstants.SOULISS_UDP_FUNCTION_POLL_RESP:
+                logger.debug("Received functional code: 0x{} - Read state answer", Integer.toHexString(functionalCode));
+                decodeStateRequest(lastByteGatewayIP, macacoPck);
+                break;
+
+            case SoulissBindingUDPConstants.SOULISS_UDP_FUNCTION_TYP_RESP:// Answer for assigned typical logic
+                logger.debug("Received functional code: 0x{}- Read typical logic answer",
+                        Integer.toHexString(functionalCode));
+                decodeTypRequest(lastByteGatewayIP, macacoPck);
+                break;
+
+            case SoulissBindingUDPConstants.SOULISS_UDP_FUNCTION_HEALTHY_RESP:// Answer
+                // nodes healthy
+                logger.debug("Received functional code: 0x{} - Nodes Healthy", Integer.toHexString(functionalCode));
+                decodeHealthyRequest(lastByteGatewayIP, macacoPck);
+                break;
+
+            case (byte) SoulissBindingUDPConstants.SOULISS_UDP_FUNCTION_DBSTRUCT_RESP:
+                logger.debug("Received functional code: 0x{} - Database structure answer",
+                        Integer.toHexString(functionalCode));
+                decodeDBStructRequest(lastByteGatewayIP, macacoPck);
+                break;
+            // case 0x83:
+            // logger.debug("Functional code not supported");
+            // break;
+            // case 0x84:
+            // logger.debug("Data out of range");
+            // break;
+            // case 0x85:
+            // logger.debug("Subscription refused");
+            // break;
+            // default:
+            // logger.debug("Unknown functional code");
+            // break;
```

Comment:
```suggestion
```
Remove commented out code.

Reply:
uncommented . must be useful for debugging souliss messages